src/pages/index.js

```diff
@@ -24,29 +26,94 @@ import {
   addModalForm,
   cardTitleInput,
   cardURLInput,
+  profileAvatarContainer,
 } from "../utils/constants.js";
 
+// Initialize API
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "c91d1d7e-9726-43c4-acb8-289510ab9429",
+    "Content-Type": "application/json",
+  },
+});
+
 // Initialize components
 
 const userInfo = new UserInfo({
   profileNameSelector: "#profile__name",
   profileJobSelector: "#profile__description",
+  avatarSelector: "#profile__pfp",
 });
 
 const profilePopup = new PopupWithForm("#edit-modal", (formData) => {
-  userInfo.setUserInfo(formData);
-  profilePopup.close();
+  profilePopup.setSaving(true);
+  api
+    .updateUserInfo(formData)
+    .then((res) => {
+      userInfo.setUserInfo({
+        name: res.name,
+        description: res.about,
+      });
+      profilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      profilePopup.setSaving(false);
+    });
 });
 profilePopup.setEventListeners();
 
 const addCardPopup = new PopupWithForm("#add-modal", (formData) => {
-  handleAddCardSubmit(formData);
+  api
```

Comment:
According to the task you need to change submit buttons text (`Saving…`) while waiting for the server response. And the user should see it.